src/linter/root.go

```diff
@@ -833,63 +838,115 @@ func (d *RootWalker) reportPhpdocErrors(n node.Node, errs phpdocErrors) {
 	}
 }
 
-func (d *RootWalker) parsePHPDocClass(doc string) classPhpDocParseResult {
-	var result classPhpDocParseResult
+func (d *RootWalker) parsePHPDocClassMethod(result *classPhpDocParseResult, part phpdoc.CommentPart) {
```

Comment:
I would write separate (unit?) tests for this piece of functionality as it seems complex enough and it is relatively self-contained.

Reply:
Done.